### ISSUE TITLE
SignalR Redis scaleout support for HA solution

### ISSUE THREAD

**Issue by cadams93** [SignalR/SignalR#3943]
As per #3564, I am confused as to the lack of support for Redis Cluster.

How is an application supposed to be highly available with Redis scaleout?

As far as I can tell:
- Sentinel is partially supported, as long as the Signalr connection is recreated on failover (getting READONLY exceptions from Redis as it tries to write to the newly promoted master??)
- Redis Cluster is not supported at all, as per documentation

Is there any plan to support a highly available solution like this in the future?

**Comment by moozzyk** [comment]
I believe https://github.com/SignalR/SignalR/issues/3564 is fixed in 2.2.2 as a duplicate of https://github.com/SignalR/SignalR/issues/3093. 
There are no plans to support Redis Cluster in 2.x at this point. Currently Redis Scaleout relies on invoking events from a lua script. When using Redis Clusters scripts are replicated to all the nodes and executed there which results in duplicate notifications which in turn break assumptions made in SignalR Redis Scalout and cause the same message to be repeated indefinitely. From what I looked fixing this would basically require redesigning and rewriting most of SignalR Redis Scaleout.

**Comment by cadams93** [comment]
Thanks for the response. I'm glad that issue with Sentinel has been resolved (will try again today - I remember testing with 2.2.0)! 

I see the predicament with Redis Cluster, and can confirm duplication of messages was the behaviour I saw (I cheekily tried this despite the documentation on the off chance the doco was not up to date). I seem to also recall a discussion - potentially on the Redis repo - describing a solution to this using bloom filters to essentially add awareness of the potential for a connected subscriber at a node level.

I guess I'll just have to wait for a solution to turn up somewhere down the line, and use Sentinel in the meantime.

Thanks :D
